packages/expo/src/launch/registerRootComponent.tsx

```diff
@@ -1,14 +1,29 @@
 import 'expo/build/Expo.fx';
+
 import * as React from 'react';
 import { AppRegistry, Platform } from 'react-native';
 
-import withExpoRoot from './withExpoRoot';
-import { InitialProps } from './withExpoRoot.types';
+type InitialProps = {
+  exp: {
+    notification?: any;
+    manifestString?: string;
+    [key: string]: any;
+  };
+  shell?: boolean;
+  shellManifestUrl?: string;
+  [key: string]: any;
+};
 
 export default function registerRootComponent<P extends InitialProps>(
   component: React.ComponentType<P>
 ): void {
-  AppRegistry.registerComponent('main', () => withExpoRoot(component));
+  if (process.env.NODE_ENV === 'production') {
+    AppRegistry.registerComponent('main', () => component);
+  } else {
+    const { withDevTools } = require('./withDevTools') as typeof import('./withDevTools');
```

Comment:
Is the "as typeof" stuff necessary? Can't tsc infer it?

Reply:
not with require syntax.